src/utils/filter-data-attributes.ts

```diff
@@ -0,0 +1,19 @@
+const regexpData = /^data-/;
+
+type PropsToBeFiltered = {
+  [key: string]: any;
+};
+export default function filterDataAttributes(
+  obj: PropsToBeFiltered
+): Partial<PropsToBeFiltered> {
+  return Object.keys(obj)
+    .filter(p => regexpData.test(p))
+    .reduce(
+      (acc, p) => {
+        // eslint-disable-next-line no-param-reassign
+        acc[p] = obj[p];
+        return acc;
+      },
+      {} as PropsToBeFiltered
+    );
+}
```

Comment:
Gradual migration with `noImplicitAny` is really hard. Anytime TypeScript can't properly infer a type it will implicitly set it to `any`. That entails that any file consumed by a typed file may have implicit any breaking type checking.

Reply:
Yeah I guess we can also allow implicit `any` as we migrate and then turn it on once we're finished